docs/source/advanced/loops.rst

```diff
@@ -0,0 +1,399 @@
+.. _loop_customization:
+
+Loop Customization
+==================
+
+Loop customization is an experimental feature introduced in Lightning 1.5 that enables advanced users to write new training logic or modify existing behavior in Lightning's training-, evaluation-, or prediction loops.
```

Comment:
nit: the `-` notation is not very commonly known, I wouldn't put it in docs

Reply:
ok will remove, I'm not sure if it's correct without them.